src/plugins/data/server/kql_telemetry/route.ts

```diff
@@ -24,6 +24,12 @@ export function registerKqlTelemetryRoute(
     .addVersion(
       {
         version: KQL_TELEMETRY_ROUTE_LATEST_VERSION,
+        security: {
+          authz: {
+            enabled: false,
+            reason: 'This route is opted out from authorization',
```

Comment:
Could we please elaborate on the reasoning there?

Reply:
but not all users are able to call this endpoint right?

Reply:
I don't want to mislead so maybe @lukasolson can confirm, but I believe they can. It's called whenever a user changes their filter language preference in Unified Search from anywhere in Kibana, which doesn't have any privileges around it: 
![CleanShot 2024-11-22 at 15 47 34@2x](https://github.com/user-attachments/assets/ad48851b-9c77-4511-ab9b-35ac2f4b5509)

Reply:
This endpoint needs to be removed. This is the sort of thing I think we were trying to catch with this PR to begin with - we are using the internal Kibana user to create a saved object when any user invokes this endpoint, which is not ideal. I'll open up a follow-up issue for this

Reply:
Opened https://github.com/elastic/kibana/issues/202518.